### PR TITLE
Added shortkey and two new options

### PR DESCRIPTION
**Pull request by kiwi0fruit**
Added shortkey: `ctrl+alt+f` and two new options:

* whether to replace text
* whether to append file name (or file path?) to the command

### CODE REVIEW

**Comment by kiwi0fruit** [comment]
Haven't tested on Linux though...

**Comment by lierdakil** [comment]
build.txt does not describe the build procedure correctly. The more or less correct version would be
```
npm install --only=dev
node_modules/.bin/tsc --project .
```

Also, appending filename to the end of the command is not a good idea. I think using OS shell's environment variable expansion might be a more robust approach.

File: package.json [hunk 1]
```diff
@@ -4,6 +4,9 @@
   "version": "0.0.1",
   "description": "A simple atom package to run a UNIX filter on current text",
   "keywords": [],
+  "activationCommands": {
+    "atom-workspace": ["unix-filter:run"]
```
Comment (lierdakil) [inline package.json]: There are two available commands though, so I reckon both should be listed here.

File: src/main.ts [hunk 2]
```diff
@@ -85,7 +93,15 @@ async function run(editor: TextEditor) {
 async function customCommand(editor: TextEditor, cmd: string) {
   const text = editor.getText()
   return new Promise<void>((resolve) => {
-    const proc = CP.exec(cmd, { encoding: 'utf8' }, (error, result) => {
+    const appendFileName = atom.config.get('unix-filter.appendFileName', {
+      scope: editor.getRootScopeDescriptor(),
+    })
+    if (appendFileName) {
+      var cmd_plus = cmd + ' "' + editor.getPath() + '"';
```
Comment (lierdakil) [inline src/main.ts]: To elaborate, this doesn't seem useful in general.

A more useful approach could be to let users use e.g. `%FILE%` environment variable via shell substitution (the name should be general enough so that most programs won't use it directly). You can override the environment by using something to the tune of
```ts
const newEnv = Object.assign({}, process.env)
newEnv.FILE = editor.getPath()
// ...
CP.exec(cmd, { env: newEnv, /*...*/ } /*...*/)
```

File: src/main.ts [hunk 3]
```diff
@@ -12,6 +12,14 @@ export const config = {
     type: 'boolean',
     default: false,
   },
+  replaceText: {
+    type: 'boolean',
+    default: false,
+  },
+  appendFileName: {
+    type: 'boolean',
+    default: true,
```
Comment (lierdakil) [inline src/main.ts]: Defaults are the opposite of what I would expect. The name and description of the package both suggest that it works as a unix filter, i.e. takes input on stdin and produces output on stdout. Neither passing additional arguments nor ignoring stdout should be the default.

Comment (kiwi0fruit) [reply src/main.ts]: Yep, the defaults in final PR should be the opposite.

File: src/main.ts (hunk 2 above)
Comment (kiwi0fruit) [reply src/main.ts]: Nice idea. I wonder if it needs some special argument like `shell=True` so that the substitution would actually happen.

File: package.json (hunk 1 above)
Comment (kiwi0fruit) [reply package.json]: Why are they here after all? I don't know why... but I added this one in case short key needs it. Why add the second one?

File: src/main.ts (hunk 2 above)
Comment (lierdakil) [reply src/main.ts]: exec already executes command in shell.
>Spawns a shell then executes the command within that shell

File: package.json (hunk 1 above)
Comment (lierdakil) [reply package.json]: See here: https://flight-manual.atom.io/hacking-atom/sections/package-word-count/#packagejson

TL;DR Atom won't activate the package until one of `activationCommands` is executed.

File: package.json [hunk 4]
```diff
@@ -4,6 +4,11 @@
   "version": "0.0.1",
   "description": "A simple atom package to run a UNIX filter on current text",
   "keywords": [],
+  "activationCommands": {
+    "atom-workspace": [
+      "unix-filter:run"
```
Comment (lierdakil) [inline package.json]: This wasn't addressed. Also, the commands are defined on `atom-text-editor`, not on `atom-workspace`.

File: src/main.ts [hunk 5]
```diff
@@ -12,6 +12,10 @@ export const config = {
     type: 'boolean',
     default: false,
   },
+  replaceText: {
+    type: 'boolean',
+    default: false,
```
Comment (lierdakil) [inline src/main.ts]: I would expect the default to be `true` here for the reasons explained before.

**Comment by kiwi0fruit** [comment]
So I've fixed everything addressed.

Works like charm by the way. I use it as [Pandoctools](https://github.com/kiwi0fruit/pandoctools) runner: https://github.com/kiwi0fruit/pandoctools/blob/master/atom.md#unix-filter

**Comment by lierdakil** [comment]
Cool. Thanks. I'll try to push out a release in a bit.

**Comment by lierdakil** [comment]
In the future, do try to write commit messages that are a little bit descriptive, it makes writing the changelog a little bit easier.

**Comment by kiwi0fruit** [comment]
I guess the hidden %FILE% ($FILE) env var needs a description somewhere.

сб, 26 мая 2018 г., 1:53 Nikolay Yakimov <notifications@github.com>:

**Comment by lierdakil** [comment]
I've added that to changelog. So it's "somewhere". Really not sure if it's worth it writing proper docs at this point, but feel free to make another PR or just create an issue if you think it's worth it.
